interp/builtin.go

```diff
@@ -559,10 +561,15 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 		r.setErr(returnStatus(code))
 	case "read":
 		var prompt string
+		var line []byte
+		var err error
```

Comment:
keep these vars declared further below, to ont keep as much in scope at once.

Reply:
Moved these variable declarations further down.